src/EhrgoHealth.WebService.UnitTests/TestAllergyIntolerance.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Linq;
+using Xunit;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Rest;
+using Newtonsoft.Json.Linq;
+using System.IO;
+
+
+namespace EhrgoHealth.WebService.UnitTests
+{
+    public class TestAllergyIntolerance
+    {
+        JObject allergyIntoleranceJSON;
+        FhirClient fhirClient;
+
+        //XUnit uses constructor for [SetUp]
+        public TestAllergyIntolerance()
+        {          
+            allergyIntoleranceJSON = JObject.Parse(File.ReadAllText(@"..\..\AllergyIntolerance.json"));
+
+            //Test server with public data. I plan to MOQ this with controlled data later
+            //As this test will cause tests to fail if the Admin purges the data on their FHIR server
+            fhirClient = new FhirClient("http://fhirtest.uhn.ca/baseDstu2/");
+        }
+
+        //Parses our own wellformed AllergyIntolerance.json
+        //I believe this is useful for having a known-good controlled JSON file of AllergyIntolerance
+        //to easily see "real" example data compared to the abstract in the documentation.
+        [Fact]
+        public void TestManualAllergyIntoleranceCode()
+        {            
+            var allergyCodeIterator =
+             from p in allergyIntoleranceJSON["substance"]["coding"]
+             select (string)p["code"];
+
+            Assert.Equal("Z88.5", allergyCodeIterator.First<String>());
+        }
+
+        //Grabs deserializes the AllergyIntolerance resource into an object
+        //Test to make sure we can grab the allergy intolerance code.
+        //This is real data from a real FHIR server at the moment, I do plan on using MOQ
+        [Fact]
+        public void TestFHIRAllergyIntoleranceCode()
+        {         
+            var allergyResource = fhirClient.Read<AllergyIntolerance>("AllergyIntolerance/6140");

```

Comment:
This was meant to test FHIR API calls to be used in the **WebServices** project, to ensure that we could connect to it from within our project, that it fetched data it was supposed to, and asserted data pulled from a real remote server that contained the AllergyIntolerance substance code. It served as working example for exactly what we intended to fetch for the decision logic. You are correct that I'm not testing the concrete classes, because it is in a separate dev branch that I haven't finished as I was still deciding allergies we will support and the medication logic I discussed yesterday.

I'll delete this and resubmit the pull request when I finish implementing the decision logic then.